File: poc-kronos-frontend/src/model/funcionario/ConfiguracaoPonto.js
class ConfiguracaoPonto {
  constructor(dados) {
    this.id = dados ? dados.id : null;
    this.idFuncionario = dados ? dados.idFuncionario : null;
    this.tarefaOnline = dados ? dados.tarefaOnline : null;
    this.horaInicialP1 = dados ? dados.horaInicialP1 : null;
    this.horaFinalP1 = dados ? dados.horaFinalP1 : null;
    this.horaInicialP2 = dados ? dados.horaInicialP2 : null;
    this.horaFinalP2 = dados ? dados.horaFinalP2 : null;
    this.horaInicialP3 = dados ? dados.horaInicialP3 : null;
    this.horaFinalP3 = dados ? dados.horaFinalP3 : null;
    this.dataInicialVigencia = dados ? dados.dataInicialVigencia : null;
    this.dataFinalVigencia = dados ? dados.dataFinalVigencia : null;
    this.cargaHorariaSemanal = dados ? dados.cargaHorariaSemanal : null;
    this.tolerancia = dados ? dados.tolerancia : null;
    this.qtdPonto = dados ? dados.qtdPonto : null;
  }

  static validations(required, numeric, dataFimMenor, horaFinalMenor) {
    return {
      idFuncionario: { required },
      tolerancia: { required },
      dataInicialVigencia: { required },
      dataFinalVigencia: {
        required,
        dataFimMenor: dataFimMenor("dataInicialVigencia")
      },
      cargaHorariaSemanal: { required, numeric },
      horaFinalP1: {
        horaFinalMenor: horaFinalMenor("horaInicialP1", true)
      },
      horaInicialP2: {
        horaFinalMenor: horaFinalMenor("horaFinalP1", true)
      },
      horaFinalP2: {
        horaFinalMenor: horaFinalMenor("horaInicialP2", true)
      },
      horaInicialP3: {
        horaFinalMenor: horaFinalMenor("horaFinalP2", true)
      },
      horaFinalP3: {
        horaFinalMenor: horaFinalMenor("horaInicialP3", true)
      }
    };
  }

  static validationMessages() {
    return {
      idFuncionario: {
        required: "Funcionário é obrigatório"
      },
      tolerancia: {
        required: "Tolerância é obrigatória"
      },
      dataInicialVigencia: {
        required: "Data Inicial Vigência é obrigatória"
      },
      dataFinalVigencia: {
        required: "Data Final Vigência é obrigatória",
        dataFimMenor:
          "Data Final de Vigência não pode ser menor que a Data Inicial de Vigência"
      },
      cargaHorariaSemanal: {
        required: "CH Semanal é obrigatória",
        numeric: "Somente número"
      },
      horaFinalP1: {
        horaFinalMenor: "Hora Final P1 não pode ser menor ou igual a Hora Inicial P1"
      },
      horaInicialP2: {
        horaFinalMenor: "Hora Inicial P2 não pode ser menor ou igual a Hora Final P1"
      },
      horaFinalP2: {
        horaFinalMenor: "Hora Final P2 não pode ser menor ou igual a Hora Inicial P2"
      },
      horaInicialP3: {
        horaFinalMenor: "Hora Inicial P3 não pode ser menor ou igual a Hora Final P2"
      },
      horaFinalP3: {
        horaFinalMenor: "Hora Final P3 não pode ser menor ou igual a Hora Inicial P3"
      }
    };
  }
}
export default ConfiguracaoPonto;

File: src/main/java/br/com/kronos/backend/aplicacao/funcionario/ConfiguracaoPonto.java
package br.com.kronos.backend.aplicacao.funcionario;

import static br.com.kronos.backend.aplicacao.validador.Validacoes.exigirNaoNulo;

import java.time.LocalDate;
import java.time.LocalTime;

import lombok.Builder;
import lombok.Getter;


@Getter
public class ConfiguracaoPonto {
	
	public static final String CACHE_NAME = "configuracoes_pontos";
	
	private Long id;
	private Long idFuncionario;
	private boolean tarefaOnline;
    private LocalTime horaInicialP1;
    private LocalTime horaFinalP1;
    private LocalTime horaInicialP2;
    private LocalTime horaFinalP2;
    private LocalTime horaInicialP3;
    private LocalTime horaFinalP3;
    private LocalDate dataInicialVigencia;
    private LocalDate dataFinalVigencia;
    private Integer cargaHorariaSemanal;
    private LocalTime tolerancia;
    
    @Builder
	public ConfiguracaoPonto(Long id, Long idFuncionario, boolean tarefaOnline, LocalTime horaInicialP1,
			LocalTime horaFinalP1, LocalTime horaInicialP2, LocalTime horaFinalP2, LocalTime horaInicialP3,
			LocalTime horaFinalP3, LocalDate dataInicialVigencia, LocalDate dataFinalVigencia,
			Integer cargaHorariaSemanal, LocalTime tolerancia) {
		this.id = id;
		this.idFuncionario = exigirNaoNulo(idFuncionario, "Funcionário");
		this.tarefaOnline = tarefaOnline;
		this.horaInicialP1 = horaInicialP1;
		this.horaFinalP1 = horaFinalP1;
		this.horaInicialP2 = horaInicialP2;
		this.horaFinalP2 = horaFinalP2;
		this.horaInicialP3 = horaInicialP3;
		this.horaFinalP3 = horaFinalP3;
		this.dataInicialVigencia = exigirNaoNulo(dataInicialVigencia, "Data Inicial Vigência");
		this.dataFinalVigencia = exigirNaoNulo(dataFinalVigencia, "Data Final Vigência");
		this.cargaHorariaSemanal = exigirNaoNulo(cargaHorariaSemanal, "Carga Horária Semana");
		this.tolerancia = exigirNaoNulo(tolerancia, "Tolerância");
	}

}
